dkobo/koboform/views/survey_draft_views.py

```diff
@@ -234,16 +307,27 @@ def publish_survey_draft(request, pk, format=None):
         status_code = 504
 
     if 'formid' in resp:
```

Comment:
this is a really long function, you can move this whole chunk to another helper function